### PR TITLE
added jv-hw-jdbc solution

### PR DESCRIPTION
**Pull request by Basilius-Third**

### CODE REVIEW

File: src/main/resources/init_db.sql [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  PRIMARY KEY (`id`)
+) ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARSET=utf8mb3;
+
+INSERT INTO `test_db`.`manufacturers` (`name`, `country`) VALUES ('Continental', 'Germany');
+INSERT INTO `test_db`.`manufacturers` (`name`, `country`) VALUES ('Michelin', 'France');
+INSERT INTO `test_db`.`manufacturers` (`name`, `country`, `is_deleted`) VALUES ('Goodyear', 'USA', '1');
+INSERT INTO `test_db`.`manufacturers` (`name`, `country`) VALUES ('Cooper', 'USA');
+INSERT INTO `test_db`.`manufacturers` (`name`, `country`) VALUES ('Yokohama', 'Japan');
+
+UPDATE manufacturers SET name = 'Basilius', country = 'Ukraine' WHERE id = 4 AND is_deleted = false;
```
Comment (PashaTerletskyi) [inline src/main/resources/init_db.sql:17]: Add empty line to the end of file

File: src/main/java/mate/jdbc/ManufacturerDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteRequest = "UPDATE manufacturers SET is_deleted = true WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteManufacturer =
+                        connection.prepareStatement(deleteRequest)) {
+            deleteManufacturer.setObject(1, id);
+            int deletedRows = deleteManufacturer.executeUpdate();
+            return deletedRows > 0;
```
Comment (PashaTerletskyi) [inline src/main/java/mate/jdbc/ManufacturerDaoImpl.java:106]:
```suggestion
            return deleteManufacturer.executeUpdate() > 0;
```

File: src/main/java/mate/jdbc/ManufacturerDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+            createManufacturer.executeUpdate();
+            ResultSet generatedKeys = createManufacturer.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                return new Manufacturer(id, name, country);
+            }
+        } catch (SQLException throwables) {
+            throw new RuntimeException("Can't create manufacturer by name: " + name, throwables);
+        }
+        return null;
```
Comment (PashaTerletskyi) [inline src/main/java/mate/jdbc/ManufacturerDaoImpl.java:36]:
```suggestion
        return manufacturer;
```

**Review by PashaTerletskyi — APPROVED** [review]

File: src/main/java/mate/jdbc/Manufacturer.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+public class Manufacturer {
+    private Long id;
+    private String name;
+    private String country;
+
+    public Manufacturer(Long id, String name, String country) {
+        this.id = id;
+        this.name = name;
+        this.country = country;
```
Comment (Sofasmile) [inline src/main/java/mate/jdbc/Manufacturer.java:11]:
```suggestion
        this(name, country);
        this.id = id;
```

File: src/main/java/mate/jdbc/Manufacturer.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
+package mate.jdbc;
```
Comment (Sofasmile) [inline src/main/java/mate/jdbc/Manufacturer.java:1]:
```suggestion
package mate.jdbc.model;
```

File: src/main/java/mate/jdbc/ManufacturerDao.java [hunk 6]
```diff
@@ -0,0 +1,16 @@
+package mate.jdbc;
```
Comment (Sofasmile) [inline src/main/java/mate/jdbc/ManufacturerDao.java:1]:
```suggestion
package mate.jdbc.dao;
```

File: src/main/java/mate/jdbc/ManufacturerDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,110 @@
+package mate.jdbc;
```
Comment (Sofasmile) [inline src/main/java/mate/jdbc/ManufacturerDaoImpl.java:1]:
```suggestion
package mate.jdbc.dao;
```

File: src/main/java/mate/jdbc/ManufacturerDaoImpl.java [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String createQuery = "INSERT INTO manufacturers (name, country) VALUES (?, ?)";
+        String name = manufacturer.getName();
+        String country = manufacturer.getCountry();
```
Comment (Sofasmile) [inline src/main/java/mate/jdbc/ManufacturerDaoImpl.java:21]: can we work without these variables: name, country. Let's use everywhere: manufacturer.getName() and manufacturer.getCountry()

File: src/main/java/mate/jdbc/ManufacturerDaoImpl.java [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createManufacturer =
+                        connection.prepareStatement(createQuery, Statement.RETURN_GENERATED_KEYS)) {
+            createManufacturer.setString(1, name);
+            createManufacturer.setString(2, country);
+            createManufacturer.executeUpdate();
+            ResultSet generatedKeys = createManufacturer.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                return new Manufacturer(id, name, country);
```
Comment (Sofasmile) [inline src/main/java/mate/jdbc/ManufacturerDaoImpl.java:31]:
```suggestion
                return manufacturer.setId(id);
```

File: src/main/java/mate/jdbc/ManufacturerDaoImpl.java [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+            }
+        } catch (SQLException throwables) {
+            throw new RuntimeException("Can't create manufacturer by name: " + name, throwables);
```
Comment (Sofasmile) [inline src/main/java/mate/jdbc/ManufacturerDaoImpl.java:34]:
```suggestion
            throw new RuntimeException("Can't create manufacturer: " + manufacturer, throwables);
```

Comment (Sofasmile) [inline src/main/java/mate/jdbc/ManufacturerDaoImpl.java:34]: you should create your custom exception
https://mate-academy.github.io/jv-program-common-mistakes/java-JDBC/jdbc-intro/JDBC-intro_checklist.html
image

File: src/main/java/mate/jdbc/ManufacturerDaoImpl.java [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+        String getQuery = "SELECT * FROM manufacturers WHERE id = ? AND is_deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getManufacturer =
+                        connection.prepareStatement(getQuery)) {
+            getManufacturer.setObject(1, id);
+            ResultSet resultSet = getManufacturer.executeQuery();
+            if (resultSet.next()) {
+                String name = resultSet.getString("name");
+                String country = resultSet.getString("country");
+                return Optional.of(new Manufacturer(id, name, country));
```
Comment (Sofasmile) [inline src/main/java/mate/jdbc/ManufacturerDaoImpl.java:50]: let's move this logic to the private method, and then we can use this method in the get() and getAll() method
https://mate-academy.github.io/jv-program-common-mistakes/java-JDBC/jdbc-intro/JDBC-intro_checklist.html
image

File: src/main/java/mate/jdbc/ManufacturerDaoImpl.java [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+        }
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String updateRequest = "UPDATE manufacturers SET name = ?, country = ? "
+                + "WHERE id = ? AND is_deleted = false;";
+        Long id = manufacturer.getId();
+        String name = manufacturer.getName();
+        String country = manufacturer.getCountry();
```
Comment (Sofasmile) [inline src/main/java/mate/jdbc/ManufacturerDaoImpl.java:84]: same here, try to remove local varibles

File: src/main/java/mate/jdbc/ManufacturerDaoImpl.java [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateManufacturer =
+                        connection.prepareStatement(updateRequest)) {
+            updateManufacturer.setString(1, name);
+            updateManufacturer.setString(2, country);
+            updateManufacturer.setObject(3, id);
+            updateManufacturer.executeUpdate();
+            return new Manufacturer(id, name, country);
```
Comment (Sofasmile) [inline src/main/java/mate/jdbc/ManufacturerDaoImpl.java:92]:
```suggestion
            return manufacturer;
```

**Review by Sofasmile — CHANGES_REQUESTED** [review]

**Review by Sofasmile — APPROVED** [review]

**Pull request closed without merging**